src/types.rs

```diff
@@ -185,24 +183,28 @@ impl BoundedStorable for StringStorable {
     const IS_FIXED_SIZE: bool = false;
 }
 
-impl Storable for PrincipalStorable {
-    fn to_bytes(&self) -> Cow<[u8]> {
-        Cow::from(self.0.as_slice())
+#[derive(Clone)]
+pub struct ApiKey(String);
```

Comment:
I would add a custom implementation of Debug/Display to be sure that no one will just add a quick `derive` trait that would leak the key.

Reply:
Pushed a `Debug` implementation which prints the API key pattern string (`{API_KEY}`). Since it doesn't seem possible to derive `Display`, this should be sufficient to address this concern. I suppose we could include a `Display` implementation anyway if you think it's worth doing.

Reply:
> Pushed a Debug implementation which prints the API key pattern string ({API_KEY}). 

Thanks! I would also add a unit test, that the debug representation of an `ApiKey` doesn't leak the value.


> Since it doesn't seem possible to derive Display, this should be sufficient to address this concern.

Make sense!